viewing_party/main.py

```diff
@@ -0,0 +1,271 @@
+
+# create movie function
+
+def create_movie(title, genre, rating):
+    new_movie = {
+        "title": title,
+        "genre": genre, 
+        "rating": rating 
+    }
+    if title and genre and rating:
+        return new_movie
+    else:
+        return None
+    
+#adds movie to user watched
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+#adds movie to watchlist
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#moves movie from watchlist to empty watched list   
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+#Calculate the average rating of all movies in watched list
+#return the average rating
+def get_watched_avg_rating(user_data):
+    if user_data["watched"]:  
+        count = 0
+        for movie in user_data["watched"]:
+            count += movie["rating"]
+            average = count / len(user_data["watched"])
```

Comment:
This all works but could be more efficient by waiting to calculate average until after the loop has finished. 
Currently `average` is re-calculated once for each movie even though the value is not used for anything until after all movies have been counted.